src/cli/transaction.go

```diff
@@ -74,6 +76,50 @@ func decodeRawTxnCmd() *cobra.Command {
 	}
 }
 
+func encodeJSONTxnCmd() *cobra.Command {
+	return &cobra.Command{
+		Short:                 "Encode JSON transaction",
+		Use:                   "encodeJsonTransaction [file path or -]",
+		DisableFlagsInUseLine: true,
+		SilenceUsage:          true,
+		Args:                  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			jsonFilePath := args[0]
+			var jsonFile *os.File
+			var err error
+			if jsonFilePath == "-" {
+				jsonFile = os.Stdin
+				err = nil
+				jsonFilePath = "<stdin>"
+			} else {
+				jsonFile, err = os.Open(jsonFilePath)
+			}
+			if err != nil {
+				return fmt.Errorf("open file failed %s:\n %v", jsonFilePath, err)
+			}
+			var rTxn readable.Transaction
```

Comment:
Use `api.CreatedTransaction`, which has a `ToTransaction` method

Reply:
`api.CreatedTransaction` does not have the same fields though , inputs in particular . The idea is to be able to encode back a transaction resulting from `decodeRawTransaction` after being created with `createTransaction` (referring to CLI every time).

Reply:
OK. Is inputs the only difference?

Reply:
If you need these methods, don't call it `GetObject`, name it `ToTransaction`, in the same way that `CreatedTransaction` does it